### ISSUE TITLE
[Proposal] Propagate original error when no fallback was provided

### ISSUE THREAD

**Issue by WrathOfZombies** [bvaughn/react-error-boundary#159]
- `react-error-boundary` version: ^4.0.4
- `node` version: 18.15.0
- `npm` version: 9.7.2

### Problem description:

When no fallback is provided, the current behavior is to throw an `Error` stating that "react-error-boundary requires either a fallback, fallbackRender, or FallbackComponent prop".

#### Our scenario

My team is building a dashboard platform where different teams can build and maintain their own tiles. Consumers can go pick a tile and it will show up on the dashboard. Additional some tiles can render tiles inside of them. 

As a platform we wrap each tile (at any depth) with some components that track the Uncaught Errors, Performance etc. 

Should a tile barf due to some uncaught exceptions, we want to log that error and then if the parent tile has provided a fallback, we'll render that automatically (happy path) and if the parent tile hasn't provided the fallback (we want to bubble the original error up).

### Suggested solution:

Instead of throwing a new error when no fallback is provided, how about just throwing the `error` that was set on the state when we set `didCatch` to true and bubble that up? Functionally it won't make a difference but gives the parent error boundary a chance to see the original issue.

I'd be happy to send a pull request and if there are better ways to achieve the same, please let me know.

### Relevant code or config

https://github.com/bvaughn/react-error-boundary/blob/13d5cf7ab57fdb59e5407e8d76ec4b8cea21fada/src/ErrorBoundary.ts#L91

```js
<ErrorBoundary
    fallback={fallback}
    onError={(error, info) => {
      console.error(`[Boundary]`, error, info);
    }}
>
    <Component {...props} />
</ErrorBoundary>
```

**Comment by bvaughn** [comment]
I think you could use the existing context api to make this happen (with a wrapper error boundary, maybe?) i wonder if you’ve considered that approach.

Happy to look at a PR.

**Comment by WrathOfZombies** [comment]
Hmm. I haven't tried that. Let me give that a shot. Else I'll send you a PR. Top of mind, I reckon it should be a 1-line change.

**Comment by WrathOfZombies** [comment]
I tried your suggestion. Yes, it works. Made the code a little worse and now we have a context/useContext pair for each tile. That's not such a bad deal but perhaps propagation of the original error is a cleaner approach.

So here's the PR - https://github.com/bvaughn/react-error-boundary/pull/160

I chose to log the original string via console.error. Let me know if you don't want that there. I consciously chose not to wrap error in Error, in order to modify the original error in any way.
